### PR TITLE
React-Redux-BookStore | Add-Remove book Function

### PR DESCRIPTION
**Pull request by errea**
# Description

This PR is to check the basic setup required for a react-redux application. An example is for the book-store project and further also check that the Add and Remove book functionality are working with the redux-store.

## Project Requirement 

Add Book Functionality
- [x] Action Creator
- [x] Add Reducer
- [x] Action constant types
- [x] Buttons to dispatch actions
- [x] Add react-router
- [x] No styling required 

Remove Book Functionality
- [x] Action Creator
- [x] Add Reducer
- [x] Action constant types
- [x] Buttons to dispatch actions
- [x] Add react-router
- [x] No styling required 

# How Has This Been Tested?
- [x] No test

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code

### CODE REVIEW

File: book-store/src/redux/books/books.js [hunk 1]
```diff
@@ -0,0 +1,31 @@
+const ADD_BOOK = 'bookStore/books/ADD_BOOK';
+const REMOVE_BOOK = 'bookStore/books/REMOVE_BOOK';
+
+const initialState = [];
+
+const addBook = (payload) => ({
+  type: ADD_BOOK,
+  payload,
```
Comment (Davidosky007) [inline book-store/src/redux/books/books.js]: * Kindly implement the add book functionality as currently, i can't add a book please fix it as it is required for this milestone [Check project requirement as a refresher ](https://github.com/microverseinc/curriculum-react-redux/blob/main/bookstore/project_create_and_remove_book.md#:~:text=In%20your%20React%20component%20responsible%20for%20adding%20new%20books%20set%20the%20data%20inputs%20in%20the%20local%20React%20state%20(set%20title%20and%20author%20-%20NOTE%3A%20categories%20and%20comments%20are%20NOT%20part%20of%20this%20step).%20Once%20your%20new%20book%20object%20is%20ready%20to%20be%20submitted%20to%20Redux%20store%2C%20dispatch%20a%20corresponding%20action%2C%20i.e.%3A)

File: book-store/src/redux/books/books.js [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+});
+
+const removeBook = (payload) => ({
+  type: REMOVE_BOOK,
+  payload,
```
Comment (Davidosky007) [inline book-store/src/redux/books/books.js]: * Kindly implement the remove book functionality too as it is a requirement for this project

**Review by Davidosky007 — CHANGES_REQUESTED** [review]
## Hi @errea 👋, 

Good job so far!
* The repo looks professional.✔️
* Git/GitHub flow best practices have been followed✔️
* You have descriptive PR✔️
* Nice using redux-logger ✔️ 
There are some issues that you still need to work on to go to the next project but you are almost there!


## Required Changes ♻️
* I think you can make your project structure this way as it is best practice 

image
* Kindly fix the linter config files as no linter check is been made on this pull request 
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

File: book-store/src/redux/books/books.js (hunk 1 above)
Comment (errea) [reply book-store/src/redux/books/books.js]: Fixed

File: book-store/src/redux/books/books.js (hunk 2 above)
Comment (errea) [reply book-store/src/redux/books/books.js]: Fixed

**Comment by errea** [comment]
Please take a careful look at the project structure requested by the reviewer and the one on my readme file, they are same.
Screen Shot 2021-08-25 at 12 16 45 PM

**Comment by Davidosky007** [comment]
> Please take a careful look at the project structure requested by the reviewer and the one on my readme file, they are same.
> Screen Shot 2021-08-25 at 12 16 45 PM

## * In your project, you have all project files in a folder called Bookstore so that's different from the structure in your Readme file

**Pull request closed without merging**